### PR TITLE
Feature/image chat messages

### LINKED ISSUE #222: As a helper, I would like to send location and picture by chat, so that I can clarify what I have and where I am.

### PR DESCRIPTION
**Pull request by danielpgp1012**
Closes #222 
This PR allows users to share pictures and share locations using the Google Static Maps api. The chat layout was modified to include a date under the message. It was also modified to include an image and location buttons. The image button allows the user to pick an image from the gallery and send it as a message. The location button allows the user to either share his/her current location, or travel to the map page where they can specify the location they want to share. Once they select the location, a static map image is generated. When this image is clicked, the user is sent to the map page to observe this location. 

Updates: 
Extract Firebase queries to ChatUtil
Add top bar text for Map limited view. 
Fix bug related to active favor count of user in ViewModel
MapPage now has a bundle argument called intentType that will make its functionality limited to a single intent. The intents can be: Request new favor, edit favor location, share location, view location.

### CODE REVIEW

**Comment by mikanikos** [comment]
Just noticed that a recent commit could have fixed the clicking behaviour of chat messages, I was testing a previous version while reviewing.

File: app/src/main/java/ch/epfl/favo/view/tabs/MapPage.java [hunk 1]
```diff
@@ -136,17 +141,97 @@ public void onMapReady(GoogleMap googleMap) {
...
-      setupFocusedFavorListen();
+      // only when the app is firstly opened, center on my location,
+      // otherwise just return where I left before
+      if (!getViewModel().isShowObservedFavor() && intentType != 0) //
+      {
+        boolean isMarkerEditable = intentType != OBSERVE_LOCATION;
+        Marker marker = drawMarkerAndFocusOnLocation(isMarkerEditable);
+        doneButton.setOnClickListener(
+            v -> {
+              if (intentType == SHARE_LOCATION) sendLocationToChat(marker);
```
Comment (huajian1069) [inline app/src/main/java/ch/epfl/favo/view/tabs/MapPage.java:162]: I notice if I drag a marker and share a new location,  the new location is actually the old one.
You can replace with this line code to fix it, 
`if (intentType == SHARE_LOCATION) sendLocationToChat(newMarkers.get(0));`
and another change at line 269

Comment (huajian1069) [reply app/src/main/java/ch/epfl/favo/view/tabs/MapPage.java:162]: I cannot leave comments at line 269, so just leave it here: 
 `  @Override
    public void onMarkerDragEnd(Marker marker) {
      newMarkers.set(0, marker);
    }`

Comment (danielpgp1012) [reply app/src/main/java/ch/epfl/favo/view/tabs/MapPage.java:162]: Great! thanks for pointing that out

**Pull request merged**